### ISSUE TITLE
“Store structure was chaged” error after package uninstall

### ISSUE THREAD

**Issue by Grawl** [rstacruz/pnpm#350]
### pnpm version:

```bash
$ pnpm -v
0.31.0
```

### Code to reproduce the issue:

```bash
$ pnpm i -S some
$ pnpm un -S some
$ pnpm i -S another
```

### Expected behavior:

Package `another` is installed and saved to `package.json`.

### Actual behavior:

Getting error:

```
 ! The store structure was changed.
      Remove it and run pnpm again.
      More info about what was changed at: https://github.com/rstacruz/pnpm/issues/276
      TIPS:
        If you have a shared store, remove both the node_modules and the shared shore.
        Otherwise just run `rm -rf node_modules`
```

Package `another` is not installed and not saved to `package.json`.

### Additional information:

 - `node -v` prints: `v6.5.0`
- macOS Sierra

**Comment by andreypopp** [comment]
@Grawl please follow the guidelines provided in error message. If you don't use custom `store-path` setting then just do `rm -rf node_modules/`.

**Comment by Grawl** [comment]
@andreypopp should I install all packages again any time I want to uninstall one?

**Comment by andreypopp** [comment]
@Grawl no

It wasn't clear to me if `node_modules/` were populated with the same version of pnpm.

Now it looks like `pnpm uninstall` corrupts the store (by writing in the old format?). I can confirm this behaviour.

**Comment by zkochan** [comment]
oh... I know what is the problem. When uninstall modifies the `store.json`, it doesn't add the pnpm version.
https://github.com/rstacruz/pnpm/blob/master/lib/api/uninstall.js#L33

it should add the pnpm version or at least don't remove the version that is already there 
install adds the version
https://github.com/rstacruz/pnpm/blob/master/lib/api/install.js#L82

**Comment by zkochan** [comment]
By the way, this would have been caught by [static typing](https://github.com/rstacruz/pnpm/issues/347)!

**Comment by Grawl** [comment]
Oh guys seems like I am the first one who tried to uninstall something 😆

**Comment by zkochan** [comment]
Fix published in v0.31.1
